frontend/src/users/components/RegisterForm.tsx

```diff
@@ -0,0 +1,88 @@
+import { FormControl, FormLabel, Input, Button, VStack, Text } from '@chakra-ui/react';
+import { FormikProps, useFormik } from 'formik';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { GenericServerError, useRegister } from '../api/useRegister';
+
+interface FormFields {
+  email: string;
+  password: string;
+  repeatPassword: string;
+}
+export interface ErrorCode {
```

Comment:
To troche ogolnie. Chyba lepiej bylo by miec interfejs pod kazdym mozliwy blad. I trzymac to w `shared`